src/mlpack/tests/lmnn_test.cpp

```diff
@@ -441,6 +441,55 @@ BOOST_AUTO_TEST_CASE(LMNNAccuracyTest)
   BOOST_REQUIRE_CLOSE(finalAccuracy, 100.0, 1e-5);
 }
 
+// Check that accuracy while learning square distance matrix is same
+// as when we are learnig low rank matrix.
+BOOST_AUTO_TEST_CASE(LMNNLowRankAccuracyTest)
+{
+  arma::arma_rng::set_seed(2);
+
+  arma::mat dataPart1;
+  dataPart1.randn(5, 50);
+
+  arma::Row<size_t> labelsPart1(50);
+  labelsPart1.fill(0);
+
+  arma::arma_rng::set_seed(50);
+
+  arma::mat dataPart2;
+  dataPart2.randn(5, 50);
+
+  arma::Row<size_t> labelsPart2(50);
+  labelsPart2.fill(1);
+
+  // Generate ordering.
+  arma::uvec ordering = arma::shuffle(arma::linspace<arma::uvec>(0, 99, 100));
+
+  // Generate datasets.
+  arma::mat dataset = join_rows(dataPart1, dataPart2);
+  dataset = dataset.cols(ordering);
+
+  // Generate labels.
+  arma::Row<size_t> labels = join_rows(labelsPart1, labelsPart2);
+  labels = labels.cols(ordering);
+
+  LMNN<> lmnn(dataset, labels, 1);
+
+  // Learn a square matrix.
+  arma::mat outputMatrix;
+  lmnn.LearnDistance(outputMatrix);
+
+  double Accuracy1 = KnnAccuracy(outputMatrix * dataset, labels, 1);
+
+  // Learn a low rank matrix.
+  outputMatrix = arma::randu(4, 5);
+  lmnn.LearnDistance(outputMatrix);
+
+  double Accuracy2 = KnnAccuracy(outputMatrix * dataset, labels, 1);
+
+  // Keeping tolerance very high.
+  // BOOST_REQUIRE_CLOSE(Accuracy1, Accuracy2, 2.0);
```

Comment:
I was thinking of checking something more like, e.g., check that both `Accuracy1` and `Accuracy2` are greater than some small value like 50%.  The concern is that if there is a problem with the low-rank LMNN, the accuracy would end up near 0.  It's fine to leave this commented out for now; let's discuss more in #1447.